test/testUcumLhcUtils.spec.js

```diff
@@ -229,61 +229,341 @@ it("should return a message for invalid unit strings", function() {
   });
 
   it("should return 96 for a request to convert 5.33 mmol/L to mg{molybdenum}/dL weight 180.156", function() {
-    var resp5 = utils.convertUnitTo('mmol/L', 5.33, 'mg{molybdenum}/dL', false, 180.156);
+    var resp5 = utils.convertUnitTo('mmol/L', 5.33, 'mg{molybdenum}/dL', { molecularWeight: 180.156 });
     assert.equal(resp5.status, 'succeeded', resp5.status + resp5.msg);
     assert.equal(resp5.toVal.toPrecision(2), 96, resp5.toVal);
   });
 
   it("should return 5.33 for a request to convert 96 mg/dL to mmol/L weight 180.156", function() {
-    var resp5 = utils.convertUnitTo('mg/dL', 96, 'mmol/L', false, 180.156);
+    var resp5 = utils.convertUnitTo('mg/dL', 96, 'mmol/L', { molecularWeight: 180.156 });
     assert.equal(resp5.status, 'succeeded', resp5.status + resp5.msg);
     assert.equal(resp5.toVal.toPrecision(3), 5.33, resp5.toVal.toPrecision(3));
   });
 
+  it("should return 0.533 for a request to convert 96 g to mol with mw 180.156", function() {
+    var resp = utils.convertUnitTo('g', 96, 'mol', { molecularWeight: 180.156 });
+    assert.equal(resp.status, 'succeeded', resp.status + resp.msg);
+    assert.equal(resp.toVal.toPrecision(3), 0.533, resp.toVal.toPrecision(3));
+  });
+
+  it("should return 96 g for a request to convert 0.533 mol with mw 180.156", function() {
+    var resp = utils.convertUnitTo('mol', 0.533, 'g', { molecularWeight: 180.156 });
+    assert.equal(resp.status, 'succeeded', resp.status + resp.msg);
+    assert.equal(resp.toVal.toPrecision(2), 96, resp.toVal.toPrecision(2));
+  });
+
+  // Test to raise error if valence is provided despite not converting to eq
+  // e.g. converting between mol and g
+  it("should return error for a request to convert 1 mol to g with valance 2 and mw 40.08", function() {
```

Comment:
@plynchnlm Question - Currently if a valence is provided, but neither unit is an equivalent, I return an error. This behavior is in line with the existing checks for mole conversions if a mw is specified but neither unit is a mole. This is because currently the code checks if a molecularWeight is provided and assumes that the conversion has to be a mass <-> mol/eq conversion.

I do wonder if this is the correct behavior - maybe its possible that a end user decides to provide a valence but is not doing a eq conversion (see this test doing a mol to mass conversion). I raise an error if the valence is provided (to match existing mole conversion behavior), but would it be better to silently ignore the valence and handle the conversion anyways?

Reply:
Fixed to be consistent - no error is thrown if excess information is provided even if it won't be used for the conversion